lib/router/handler-info.js

```diff
@@ -24,7 +27,52 @@ function HandlerInfo(_props) {
 
 HandlerInfo.prototype = {
   name: null,
-  handler: null,
+
+  getHandler: function() {},
+
+  get handler() {
+    if (this.hasOwnProperty('_handler')) {
```

Comment:
I used `hasOwnProperty` since it might be `undefined`, which makes checks tricky. Basically we want "this has been set, but the type of value is ambiguous".